0.57 RN support

nickwinder opened the pull request:
# Details
This is PR is responsible for updating and ensuring support of react-native 0.57.
Due to changes RNFS, react-native, react and react-native-windows had to be updated. Along with this, I had to change the Windows target version for `react-native-pspdfkit` and `Catalog`. 

# Acceptance Criteria
- [x] Update to RN 0.57 and any other dependencies that are required.
- [x] The Catalog example on iOS works.
- [x] The Catalog example on Android works.
- [x] The Catalog example on Windows works.
- [x] Standalone example on iOS works.
- [x] Standalone example on Android works.

nickwinder commented:
@simoarpe @radazzouz I assume this will need changes on the Android/iOS. `ListView` has been deprecated, which is used in the `Catalog`. In windows, I've swapped this out for `FlatList` which is similar. See `index.windows.js` for example. I want to try and get this in for a customer, so let me know any blockers.

radazzouz commented:
I retested this on iOS and we get the same warnings:

<img width="658" alt="screen shot 2019-01-29 at 4 21 15 pm" src="https://user-images.githubusercontent.com/7443038/51941202-fd8cfd00-23e1-11e9-908d-3ec1b0331dea.png">

nickwinder commented:
@radazzouz ListView I can fix as I had to do the same with UWP. Any ideas on the other warning?

radazzouz commented:
> @radazzouz ListView I can fix as I had to do the same with UWP. Any ideas on the other warning?

That's because we weren't implementing `+[RCTPSPDFKitManager requiresMainQueueSetup]`. I fixed it in https://github.com/PSPDFKit/react-native/pull/162/commits/32947441d44461bdfe62d5e9e03c72c89c157803

nickwinder commented:
@simoarpe I fixed up `ListView`.
I also went through the README to build a standalone example. There were a fair few changes since `0.55.4`. I appreciate if you could give this a once over. In particular the `tasks` I removed from `app/build.gradle`, I found they were not needed. 
After this is complete, I'm am happy to merge.
